src/hooks/useBalance/useCosmosBalance.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { StargateClient } from '@cosmjs/stargate'
+import type { TUseBalanceOptions } from './types'
+import { isCosmosWallet } from '@/utils/wallet'
+import { getNetworkById, rpcMapping } from '@/networks'
+
+const SECONDS_BEFORE_NEXT_UPDATE = 10
+
+function useCosmosBalance(options: TUseBalanceOptions) {
+  const { chainId } = options
+  const isCosmos = isCosmosWallet(options)
+
+  const [balance, setBalance] = useState<string | null>(null)
+
+  const network = useMemo(() => chainId ? getNetworkById(chainId) : null, [chainId])
+  const rpcAddress = useMemo(() => chainId ? rpcMapping[chainId] : '', [chainId])
+  const clientPromise = useMemo(async () => await StargateClient.connect(rpcAddress), [rpcAddress])
```

Comment:
may throw error, if can't find network or rpc address? these fallbacks looks not intuitive